2005-08-25  Alfred M. Szmidt  <ams@gnu.org>
        
	* Makefile (MSGIDS): New variable.
	(all, install-msgids, $(datadir)/msgids): New targets.
	(install): Specify install-header and install-msgids as
	prerequisites.

---
 hurd/Makefile |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

--- a/hurd/Makefile
+++ b/hurd/Makefile
@@ -22,15 +22,20 @@ DIST_FILES = subsystems $(notdir $(hdrs)
 
 INSTHDRS = hurd_types.h version.h ioctl_types.h paths.h shared.h console.h \
 	$(notdir $(wildcard $(srcdir)/*.defs))
+MSGIDS := hurd.msgids $(patsubst %.defs,%.msgids,$(filter %.defs,$(INSTHDRS)))
 
 include ../Makeconf
 
-install-headers install: $(includedir)/hurd \
+all: $(MSGIDS)
+
+install: install-msgids install-headers
+install-headers: $(includedir)/hurd \
 	$(addprefix $(includedir)/hurd/,$(INSTHDRS))
+install-msgids: $(MSGIDS) $(datadir)/msgids; $(INSTALL_DATA) $^
 
 $(includedir)/hurd/%: $(srcdir)/%; $(INSTALL_DATA) $< $@
 
-$(includedir)/hurd:;mkdir -p $@
+$(datadir)/msgids $(includedir)/hurd:;mkdir -p $@
 
 %.msgids: $(srcdir)/%.defs
 	if grep -q '^subsystem' $<; \
